src/java/picard/sam/markduplicates/MarkDuplicates.java

```diff
@@ -109,9 +109,19 @@
     @Option(doc = "Read two barcode SAM tag (ex. BX for 10X Genomics)", optional = true)
     public String READ_TWO_BARCODE_TAG = null;
 
+    @Option(doc = "If true remove optical duplicates from the output even if REMOVE_DUPLICATES is false.")
+    public boolean REMOVE_OPTICAL_DUPLICATES = false;
+
+    @Option(doc = "If set write a tag with this name and OPTICAL_DUPLICATE_TAG_VALUE on to each read that is an optical duplicate.", optional=true)
+    public String OPTICAL_DUPLICATE_TAG_NAME = null;
+
+    @Option(doc = "If true do not write duplicates to the output file instead of writing them with appropriate flags set.")
+    public String OPTICAL_DUPLICATE_TAG_VALUE = "+";
+
     private SortingCollection<ReadEndsForMarkDuplicates> pairSort;
     private SortingCollection<ReadEndsForMarkDuplicates> fragSort;
     private SortingLongCollection duplicateIndexes;
+    private SortingLongCollection opticalDuplicateIndices;
```

Comment:
even if both of the spellings of the plural of index are correct, I suggest we use one.